File: tests/testthat/test-basic.R
context('Basic conversions')

test_that('valid iso3c to country.name works', {
    name_of <- function(iso3c_code) countrycode(iso3c_code, 'iso3c', 'country.name')
    expect_equal(name_of('CAN'), 'Canada')
    expect_equal(name_of(c('USA', 'CAN')), c('United States', 'Canada'))
})

test_that('invalid iso3c to country.name returns NA', {
    name_of <- function(iso3c_code) countrycode(iso3c_code, 'iso3c', 'country.name', warn = FALSE)
    expect_equal(name_of('BAD'), NA_character_)
    expect_equal(name_of(c('BAD', 'BLA', 'CAN')), c(NA_character_, NA_character_, 'Canada'))
})

test_that('warn=TRUE gives warnings, but does not break conversion', {
    iso2c_of <- function(cowc_code) countrycode(cowc_code, 'cowc', 'iso2c', warn=TRUE)
    expect_warning(val <- iso2c_of('BLA'), 'not matched')
    expect_equal(val, NA_character_)
    expect_warning(val <- iso2c_of(c('ALG', 'USA')), NA)
    expect_equal(val, c('DZ','US'))
    expect_warning(val <- iso2c_of(c('BLA', 'USA')), 'not matched')
    expect_equal(val, c(NA_character_,'US'))
})

test_that('warn=FALSE does not give warnings', {
    iso2c_of <- function(cowc_code) countrycode(cowc_code, 'cowc', 'iso2c', warn=FALSE)
    expect_warning(iso2c_of('BLA'), NA)
    expect_warning(iso2c_of(c('BLA', 'USA')), NA)
})

File: python/tests/test_basic.py
from countrycode import countrycode
import polars as pl

from fixture_data import CONVERSIONS_BY_CODE

"""
Test to check that finding the iso3n representation of an iso3c row is
equivalent to finding the corresponding cell in the countrycode dataframe.
"""


def test_basic_conversions():
    def name_of(iso3c_code):
        return countrycode(iso3c_code, origin="iso3c", destination="country.name")

    cases = CONVERSIONS_BY_CODE["iso3c"]["country.name"]
    source = ["usa", "CAN"]
    assert name_of(source) == [cases[code] for code in source]


def test_invalid_iso3c_to_country_name():
    def name_of(iso3c_code):
        return countrycode(iso3c_code, "iso3c", "country.name", warn=False)

    cases = CONVERSIONS_BY_CODE["iso3c"]["country.name"]
    assert name_of(["BAD", "BLA", "CAN"]) == [cases["BAD"], None, cases["CAN"]]


def test_readme_examples():
    # Example 1: Convert ISO numeric codes to French country names
    assert countrycode([12, 124], origin="iso3n", destination="cldr.short.fr") == [
        "Algérie",
        "Canada",
    ]

    assert countrycode(12, origin="iso3n", destination="cldr.short.fr") == "Algérie"

    # Example 2: Convert country names to ISO3C codes
    assert countrycode(
        ["canada", "United States", "alGeria"],
        origin="country.name.en.regex",
        destination="iso3c",
    ) == ["CAN", "USA", "DZA"]

    # Example 3: Convert using Polars Series
    result = countrycode(
        pl.Series(["DZA", "CAN", "USA"]), origin="iso3c", destination="cldr.short.fr"
    )
    assert isinstance(result, pl.Series)
    assert result.to_list() == ["Algérie", "Canada", "É.-U."]

    # Example 4: Convert ISO3C to ISO numeric
    assert countrycode(["DZA", "CAN", "USA"], origin="iso3c", destination="iso3n") == [
        12,
        124,
        840,
    ]

    # Example 5: Convert single ISO numeric to German country name
    assert countrycode(12, origin="iso3n", destination="cldr.short.de") == "Algerien"

    # Example 6: Convert long country names to ISO3C
    assert countrycode(
        ["Democratic Republic of Vietnam", "Algeria"],
        origin="country.name.en.regex",
        destination="iso3c",
    ) == ["VNM", "DZA"]

    # Example 7: Convert German country name to ISO3C
    assert countrycode(["Algerien"], origin="country.name.de", destination="iso3c") == [
        "DZA"
    ]
